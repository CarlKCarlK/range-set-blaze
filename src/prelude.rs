//! This prelude module provides a convenient way to import the most commonly used
//! types, traits, and functions.
//!
//! ```
//! use range_set_blaze::prelude::*;
//! ```
pub use crate::{
    intersection_dyn, intersection_map_dyn, union_dyn, union_map_dyn, CheckSortedDisjoint,
    DynSortedDisjoint, DynSortedDisjointMap, MultiwayRangeMapBlaze, MultiwayRangeSetBlaze,
    MultiwaySortedDisjoint, MultiwaySortedDisjointMap, RangeMapBlaze, RangeSetBlaze,
    SortedDisjoint, SortedDisjointMap, SortedStarts,
};
