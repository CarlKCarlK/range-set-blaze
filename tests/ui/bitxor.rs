use range_set_blaze::prelude::*;
use std::ops::BitXor;

fn main() {
    let a = RangeSetBlaze::from_iter([1, 2, 3]).into_ranges();
    let b = [(1, 2), (3, 4)].into_iter();
    let _c = a.bitxor(b);
}
