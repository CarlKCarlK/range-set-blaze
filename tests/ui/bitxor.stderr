error[E0599]: no method named `bitxor` found for struct `IntoRangesIter` in the current scope
 --> tests/ui/bitxor.rs:6:16
  |
6 |     let _c = a.bitxor(b);
  |                ^^^^^^ method not found in `IntoRangesIter<{integer}>`
  |
 ::: $RUST/core/src/ops/bit.rs
  |
  |     fn bitxor(self, rhs: Rhs) -> Self::Output;
  |        ------ the method is available for `IntoRangesIter<{integer}>` here
  |
  = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
1   | use core::ops::BitXor;
    |
